Source/WebCore/style/StyleBuilder.cpp

```diff
@@ -398,6 +398,9 @@ void Builder::applyProperty(CSSPropertyID id, CSSValue& value, SelectorChecker::
         return;
     }
 
+    if (!isUnset && cascadeLevel == CascadeLevel::Author && m_state.element()->isDevolvableWidget() && CSSProperty::disablesNativeAppearance(id) && m_state.applyPropertyToRegularStyle())
+        style.setNativeAppearanceDisabled(true);
+
     BuilderGenerated::applyProperty(id, m_state, valueToApply.get(), valueType);
```

Comment:
This extra adjustment would look nicer after the`BuilderGenerated::applyProperty` call since that is still the main thing happening here. It can also be combined in a since `if (!isUnset)` branch with the existing code there.